### PR TITLE
feat(JavaScript): implement xlang protocol

### PR DESCRIPTION
**Pull request by theweipeng**
## What does this PR do?
1. Remove unsigned types which are removed in xlang protocol.
2. Rename  `long int double` to `int64  int32 float64` to ensure consistency with other languages
3. Implements the float16 writer and float16 reader.
## Does this PR introduce any user-facing change?
The JavaScript implementation has not been released yet, so there are no user-facing changes.

## Benchmark
There are no fluctuations in the benchmark.

| (index)  | serialize | deserialize |
|----------|-----------|-------------|
| fury     | 116       | 118         |
| protobuf | 25        | 33          |
| json     | 20        | 23          |

### CODE REVIEW

**Comment by theweipeng** [comment]
@LiangliangSui @chaokunyang  Hi, This PR addresses the numeric part of the xlang protocol; could you please help review it? The other parts of the protocol will be submitted in another PR.

File: javascript/packages/fury/lib/meta.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ export const getMeta = (description: TypeDescription, fury: Fury): Meta => {
         needToWriteRef: Boolean(fury.config.refTracking) && false,
         type,
       };
+    case InternalSerializerType.BINARY:
+      return {
+        fixedSize: 8,
+        needToWriteRef: Boolean(fury.config.refTracking) && true,
```
Comment (chaokunyang) [inline javascript/packages/fury/lib/meta.ts:91]:
```suggestion
        needToWriteRef: Boolean(fury.config.refTracking),
```

seems `&& true` is unnecessary

File: javascript/packages/fury/lib/gen/datetime.ts [hunk 2]
```diff
@@ -72,5 +72,5 @@ class DateSerializerGenerator extends BaseSerializerGenerator {
   }
 }
 
-CodegenRegistry.register(InternalSerializerType.DATE, DateSerializerGenerator);
+CodegenRegistry.register(InternalSerializerType.DURATION, DateSerializerGenerator);
```
Comment (chaokunyang) [inline javascript/packages/fury/lib/gen/datetime.ts:75]:
```suggestion
CodegenRegistry.register(InternalSerializerType.DURATION, DurationSerializerGenerator);
```

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**